### PR TITLE
SEVIRI LST datareader

### PR DESCRIPTION
**Pull request by MWesselkamp**
## Description

I implemented a custom datareader for the SEVIRI Level 2 land surface temperature product (LST). It works on on a gridded (lat lon) datastore of full Seviri images at hourly temporal resolution. The datastore is loaded with xarray during initialisation and lazily in the workers to avoid deadlocks caused by forked dataloaders. 

The datareader extracts regions of interest from an image using indices that are stored in metadata files.  It subsamples in time and space, e.g. here reduces hourly data to 6-hourly time steps. It loads normalisation statistics from external metadata files. It incorporates geoinfo (e.g. here DEM) from the datastore or returns empty geoinfo if not specified in the config.

This datareader is added to the MultiStreamSampler. A config used for finetuning for prognostic forecasting is added to configs/streams/.

I documented the datareader implementation with medium level of detail here: https://gitlab.jsc.fz-juelich.de/hedgedoc/s/LvbrmAWvf#

## Issue Number

This is not linked to an issue.

## Checklist before asking for review

-   [x] I have performed a self-review of my code
-   [x] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [x] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [x] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

**Comment by clessig** [comment]
@MWesselkamp : could rebase to the latest develop (or open a new PR with your work based on develop). If you just worked on the data reader, it shouldn't be an issue to rebase wrt develop.

**Comment by MWesselkamp** [comment]
I have created a new PR [#1836](https://github.com/ecmwf/WeatherGenerator/pull/1836) rebased onto latest develop as requested. Closing this one in favor of the rebased version.
